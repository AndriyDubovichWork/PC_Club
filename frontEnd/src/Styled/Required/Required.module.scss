.content {
  display: flex;
  flex-direction: row;
}
.Required {
  color: red;
}
