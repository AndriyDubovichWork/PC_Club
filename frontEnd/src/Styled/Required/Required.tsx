import style from './Required.module.scss';

const Required = ({ text }: { text: string }) => {
  return (
    <div className={style.content}>
      <p>{text}</p>
      <p className={style.Required}>*</p>
    </div>
  );
};

export default Required;
