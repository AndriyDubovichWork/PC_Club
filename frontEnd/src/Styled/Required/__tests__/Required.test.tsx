import { render, screen, cleanup } from '@testing-library/react';
import Required from '../Required';
import renderer from 'react-test-renderer';
import { MemoryRouter, Route, Routes } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

test('Required match snapshot', () => {
  const tree = renderer
    .create(
      <MemoryRouter initialEntries={['/PC_Club/']}>
        <Routes>
          <Route path='/PC_Club/' element={<Required text='hi' />} />
        </Routes>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('Required contains text', () => {
  render(
    <MemoryRouter initialEntries={['/PC_Club/']}>
      <Routes>
        <Route path='/PC_Club/' element={<Required text='hi' />} />
      </Routes>
    </MemoryRouter>
  );

  expect(screen.getByText('hi')).toBeInTheDocument();
});
