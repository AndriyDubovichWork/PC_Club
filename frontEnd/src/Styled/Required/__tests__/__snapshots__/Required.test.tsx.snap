// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Required match snapshot 1`] = `
<div
  className="content"
>
  <p>
    hi
  </p>
  <p
    className="Required"
  >
    *
  </p>
</div>
`;
